README.md

```diff
@@ -9,3 +9,65 @@ See [Declarative String Processing Overview][decl-string]
 ## Requirements
 
 - [Swift Trunk Development Snapshot](https://www.swift.org/download/#snapshots) DEVELOPMENT-SNAPSHOT-2022-02-03 or later.
+
+## Integration with Swift
+
+`_MatchingEngine`, `_CUnicode` and `_StringProcessing` are specially integrated modules that are built as part of apple/swift.
+
+Specifically, `_MatchingEngine` contains the parser for regular expression literals and is built both as part of the compiler and as a core library. `_CUnicode` and `_StringProcessing` are built together as a core library named `_StringProcessing`.
+
+| Module              | Swift toolchain component                                                            |
+| ------------------- | ------------------------------------------------------------------------------------ |
+| `_MatchingEngine`   | `SwiftCompilerSources/Sources/ExperimentalRegex` and `stdlib/public/_MatchingEngine` |
+| `_CUnicode`         | `stdlib/public/_StringProcessing`                                                    |
+| `_StringProcessing` | `stdlib/public/_StringProcessing`                                                    |
+
```

Comment:
I think this is good for an initial integration doc, but I do think we'll want some CMake, perhaps in the Swift repo, that would add or remove certain sub-folders for the stdlib, like the engine itself.